### PR TITLE
Add reason for receiving a notification to workflow emails.

### LINKED ISSUE #4305: Fix unsubscribe link
It looks tacky af. It's like we're saying "the call to action in every email is to unsubscribe".

### PR DESCRIPTION
**Pull request by tkaemming**
This should clarify some confusion that users have around why they are receiving some emails.

This also fixes GH-4305.

### CODE REVIEW

File: src/sentry/web/frontend/debug/mail.py [hunk 1]
```diff
@@ -123,11 +123,18 @@ def render(self, request):
...
+        self.request = request
         self.email = emails.get(activity.type)(activity)
 
     def get_context(self):
         context = self.email.get_base_context()
+        context.update({
+            'reason': get_random(self.request).choice(
+                GroupSubscriptionReason.descriptions.values()
+            ),
+            'unsubscribe_link': 'javascript:alert("This is a preview page, what did you expect to happen?");',
```
Comment (tkaemming) [inline src/sentry/web/frontend/debug/mail.py]: 😎

File: src/sentry/models/groupsubscription.py [hunk 2]
```diff
@@ -6,18 +6,28 @@
...
 
 from sentry.db.models import (
-    BoundedPositiveIntegerField, FlexibleForeignKey, Model, BaseManager,
+    BaseManager, BoundedPositiveIntegerField, FlexibleForeignKey, Model,
     sane_repr
 )
 
 
 class GroupSubscriptionReason(object):
+    implicit = -1  # not for use as a persisted field value
```
Comment (tkaemming) [inline src/sentry/models/groupsubscription.py]: Not thrilled with this, but figured it was better than adding a random constant value elsewhere. The `CHECK` constraint on the database should prevent this from being inserted.

File: src/sentry/models/groupsubscription.py [hunk 3]
```diff
@@ -6,18 +6,28 @@
...
     unknown = 0
     comment = 1
     assigned = 2
     bookmark = 3
     status_change = 4
 
+    descriptions = {
+        implicit: u"have opted to receive updates for all issues within "
+                  "projects that you are a member of",
```
Comment (tkaemming) [inline src/sentry/models/groupsubscription.py]: I couldn't think of a good way to word this, this explains accurately but reads a little weird, suggestions accepted.

File: src/sentry/plugins/sentry_mail/activity/base.py [hunk 4]
```diff
@@ -246,13 +243,19 @@ def send(self):
...
             if group:
-                context['unsubscribe_link'] = generate_signed_link(
-                    user.id,
-                    'sentry-account-email-unsubscribe-issue',
-                    kwargs={'issue_id': group.id},
-                )
+                context.update({
+                    'reason': GroupSubscriptionReason.descriptions.get(
+                        reason,
+                        "are subscribed to this issue",
```
Comment (tkaemming) [inline src/sentry/plugins/sentry_mail/activity/base.py]: This seemed like a reasonable enough default?

File: src/sentry/templates/sentry/emails/activity/generic.html [hunk 5]
```diff
@@ -4,14 +4,28 @@
...
+
+    <h2>{{ activity_name }}</h2>
+
+    <p>{{ html_description }}</p>
+
+  {% endblock %}
 
   {% include "sentry/emails/group_header.html" %}
+
+  {% if reason %}
```
Comment (tkaemming) [inline src/sentry/templates/sentry/emails/activity/generic.html]: This conditional shouldn't be necessary, but added it just to be safe and ensure this didn't result in a sentence fragment.,

**Review by tkaemming — COMMENTED** [review]
These changes are primarily visual, so it's probably a good idea to check out the visual diffs.

File: src/sentry/models/groupsubscription.py (hunk 2 above)
Comment (dcramer) [inline src/sentry/models/groupsubscription.py]: yeah not great, but as long as its commented it seems fine

**Review by dcramer — APPROVED** [review]

**Pull request merged**